### DISCUSSION TITLE
Replication delay due to a missing attachments loop > Raven.Server.Exceptions.MissingAttachmentException: Revision

### DISCUSSION THREAD

**Discussion by oiBio** [ravendb/ravendb#16624] [Q&A] [answered]
I was playing a bit with External Replication and it got stuck with this error.

I was wondering two things. How does it happen, that the Attachment is not found anymore? And how can i delete the Revision?
I can in fact not access the Attachment of the Revision in given document  'RavenDbFile/2510-B'
 
> Replication delay due to a missing attachments loop
> 
> Failed to send batch successfully to [XXXX @ https://YYY]. Destination reported missing attachments 2 times.
> Please try to delete the missing attachment from 'XXXX ' on node B (see additional information regarding the document and attachment below)
> 
> ```
> Raven.Server.Exceptions.MissingAttachmentException: Revision 'RavenDbFile/2510-B' with change vector 'B:84079-mfuFomuyzUS+edXcvvHXdQ' has attachment named: 'image.png', hash: 'xW/IlwbVwZ8SdVaVapT6QjWlfqjzS6tgKfD7amSCj2s=' listed as one of its attachments but it doesn't exist in the attachment storage
>    at Raven.Server.Documents.Replication.IncomingReplicationHandler.MergedDocumentReplicationCommand.AssertAttachmentsFromReplication(DocumentsOperationContext context, DocumentReplicationItem doc) in C:\Builds\RavenDB-Stable-5.4\54049\src\Raven.Server\Documents\Replication\IncomingReplicationHandler.cs:line 1626
>    at Raven.Server.Documents.Replication.IncomingReplicationHandler.MergedDocumentReplicationCommand.ExecuteCmd(DocumentsOperationContext context) in C:\Builds\RavenDB-Stable-5.4\54049\src\Raven.Server\Documents\Replication\IncomingReplicationHandler.cs:line 1302
>    at Raven.Server.Documents.TransactionOperationsMerger.ExecutePendingOperationsInTransaction(List`1 executedOps, DocumentsOperationContext context, Task previousOperation, DurationMeasurement& meter) in C:\Builds\RavenDB-Stable-5.4\54049\src\Raven.Server\Documents\TransactionOperationsMerger.cs:line 942
>    at Raven.Server.Documents.TransactionOperationsMerger.MergeTransactionsOnce() in C:\Builds\RavenDB-Stable-5.4\54049\src\Raven.Server\Documents\TransactionOperationsMerger.cs:line 514
> --- End of stack trace from previous location ---
>    at Raven.Server.Documents.TransactionOperationsMerger.Enqueue(MergedTransactionCommand cmd) in C:\Builds\RavenDB-Stable-5.4\54049\src\Raven.Server\Documents\TransactionOperationsMerger.cs:line 144
> ```

**Comment by shiranshalom** [comment]
Hello @oiBio , 
Can you provide the version you are working on?
It is a known issue recently fixed (you can see more details [here](https://issues.hibernatingrhinos.com/issue/RavenDB-19421/Missing-attachments-in-Replication)). 
In general, the error indicates that an attachment reference exists in Revision '_RavenDbFile/2510-B_' metadata, but the actual data is missing; you can delete the revision using the following endpoint:  `/databases/*/admin/revisions` , for example, via Postman: 
`DELETE` `https://YYY[/databases/XXXX/admin/revisions`
`{"DocumentIds":["RavenDbFile/2510-B"]}`

Hope it helps, 
please keep us posted and feel free to contact us again if you have more questions :-)

**Reply by ayende** [reply]
To give more context, the behavior you are seeing is there to prevent a scenario where due to a problem with attachments (shouldn't happen, that is the fixed bug @shiranshalom is talking about) you are now in a replication loop.
Always sending the same batch over & over again. That is a bad idea, since this may burn through a lot of bandwidth traffic, and in the cloud you are charged for that.

**Comment by oiBio** [comment]
Hi Thanks you for our response.
I use the current Version 5.4.104 and i can also can replicate this behavior. The record itself (without the revisions) is ~2 Months old, i cant remember the exact version but i think it was 5.3.103 or 5.4.5 (not sure)
I have a "Template" Database for our application and i need to copy this.
I did the following

1. Create New Database
2. In Studio > Tasks > Import Data > From RavenDB Server
3. From current Database and select Template database
4. The attachment in the current file workes fine (the record does not have any revisions yet)
5. Activate Revision via Document Defaults (these Settings does not get transfered with Import "From RavenDB Server")
6. Delete the attachment in document "RavenDbFile/2510-B"
7. Voila! No you have the missing attachment in the revision

I extracted the document so you can reproduce this as well

- Just import the data from this .ravendbdump [Dump of Artemis.TEST 2023-06-09 10-58.ravendbdump.zip](https://github.com/ravendb/ravendb/files/11704401/Dump.of.Artemis.TEST.2023-06-09.10-58.ravendbdump.zip)
- activate revisions via document defaults
- delete attachment from document
- you cant download the attachment from the revision

I am pretty sure you would get the same error when you retry the external replication, but i think here lays the error

Hope this helps :-)

**Reply by garayx** [reply, marked as answer]
Thanks for the reproduce, we will check this out,
https://issues.hibernatingrhinos.com/issue/RavenDB-20684
